src/SourceLocator/Type/SingleFileSourceLocator.php

```diff
@@ -43,9 +44,28 @@ public function __construct(private string $fileName, Locator $astLocator)
      */
     protected function createLocatedSource(Identifier $identifier): LocatedSource|null
     {
+        $content = file_get_contents($this->fileName);
+        if (! $content) {
+            return null;
+        }
+
+        $name = $identifier->getName();
+
+        if (
+            $name !== Identifier::WILDCARD &&
+            ! str_starts_with($name, ReflectionClass::ANONYMOUS_CLASS_NAME_PREFIX)
+        ) {
+
+            $shortName = array_reverse(explode('\\', $identifier->getName()))[0];
+
+            if (! str_contains($content, $shortName)) {
+                return null;
+            }
+        }
+
         return new LocatedSource(
-            file_get_contents($this->fileName),
-            $identifier->getName(),
+            $content,
+            $name,
             $this->fileName,
         );
     }
```

Comment:
We should probably attempt filtering in an alternative implementation of `SingleFileSourceLocator`, then look for who is instantiating `SingleFileSourceLocator`, and see if we can somehow inject the constructor for it there :thinking: 

It's just an idea, but it would allow for filtering and transforming sources before skipping or passing them to parsing/reflection